### ISSUE TITLE
 There used to be views and downloads on this page. 

### ISSUE THREAD

**Issue by kjgarza** [datacite/lupo#360]
Currently the API is returning 0 views and 0 downloads. It also happens when calling graphql. Example:
```
{
  dataset(id:“10.7272/q6g15xs4”) {
    id
    titles{
      title
    }
    viewCount
    downloadCount
    citationCount
  }
}
```
This might be related to the list of breaking changes from `ElasticSearch@7`: https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html#breaking_70_scripting_changes
For example
> getMillis() should be replaced with toInstant().toEpochMilli()
Its used in aggregations.

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_146tf)

**Comment by kjgarza** [comment]
The problem is a bit more complex but not that bad.

The reason why we get 0 usage and citations is that there are actually 0 events that follow the filtering criteria. 

The criteria is not wrong but the wya EventData is indexing events is. So all the index mapping is done with `camelCase` however all event proporties are converted to kebab case in the controller.

here:
https://github.com/datacite/lupo/blob/d875b92346f24361d95e8ab445a326ab83dc254f/app/controllers/events_controller.rb#L237

On can see that in kibana how the values in the index are in the kebaba case version of `date_published` but not in `datePublished` which is the one in the ES mapping.
https://8cf39df5abd84a29b63175bdce2ba1c8.eu-west-1.aws.found.io/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-5h,to:now))&_a=(columns:!(obj.date_published,source_id,obj.id),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'9b656b20-0ea1-11ea-99a9-b129f692045f',key:source_id,negate:!f,params:(query:datacite-usage),type:phrase,value:datacite-usage),query:(match:(source_id:(query:datacite-usage,type:phrase))))),index:'9b656b20-0ea1-11ea-99a9-b129f692045f',interval:m,query:(language:kuery,query:''),sort:!(!(updated_at,desc),!(source_id,desc)))

![image](https://user-images.githubusercontent.com/1092861/69738939-229fe880-1137-11ea-9ab5-09f6d6b0d903.png)




This was not failing before because Es@6 won't throw an expection if the field was empty, but the new version it would.

Basically this is related to bug https://github.com/datacite/lupo/issues/352


Potential solutions to make the query

change the whole mapping and reindex all the events (it doesn need to reimport) <- because in the database everythin is saved with snacke_case

**Comment by kjgarza** [comment]
usage is back up https://search.datacite.org/works?query=prefix%3A10.7272

**Comment by kjgarza** [comment]
![image](https://user-images.githubusercontent.com/1092861/70043037-2bbe0900-15c0-11ea-9903-dffc743b158b.png)
